### PR TITLE
Cleanup and testing

### PR DESCRIPTION
**Pull request by buehner**
The past weeks, a lot of coding was done. I started some cleanup and test-writing.

### CODE REVIEW

File: src/shogun2-core/shogun2-service/src/main/java/de/terrestris/shogun2/service/ImageFileService.java [hunk 1]
```diff
@@ -58,46 +58,79 @@ public void setDao(D dao) {
...
+
+	/**
+	 * @param uploadedImage
+	 * @return
+	 * @throws Exception
+	 */
+	public ImageFile uploadImageFile(MultipartFile uploadedImage) throws Exception {
+
+		if (uploadedImage == null || uploadedImage.isEmpty()) {
+			final String errMsg = "Upload failed. Image " + uploadedImage + " is empty.";
```
Comment (dnlkoch) [inline src/shogun2-core/shogun2-service/src/main/java/de/terrestris/shogun2/service/ImageFileService.java]: `uploadedImage` can potentially be `null` here.

Comment (marcjansen) [inline src/shogun2-core/shogun2-service/src/main/java/de/terrestris/shogun2/service/ImageFileService.java]: I think concatenation will nonetheless work

Comment (dnlkoch) [inline src/shogun2-core/shogun2-service/src/main/java/de/terrestris/shogun2/service/ImageFileService.java]: It's not about a NullPointer, it's about the clarity of the errorMessage. But still minor².

Comment (buehner) [inline src/shogun2-core/shogun2-service/src/main/java/de/terrestris/shogun2/service/ImageFileService.java]: NullPointer will never happen here as the evaluation of `if (uploadedImage == null || uploadedImage.isEmpty())` would stop after the first check if uploadedImage would be null :wink:

**Comment by buehner** [comment]
This is ready for merging now! Please review one more time and feel free to merge

**Comment by weskamm** [comment]
Looks good to me, will merge!

**Pull request merged**